### ISSUE TITLE
HTTP::Error Unknown MIME type: text/html

### ISSUE THREAD

**Issue by tak1n** [api-ai/apiai-ruby-client#18]
When API.ai does not respond with json calling `#parse` on a HTTP response raises an exception as no other Mime Type handler is registered yet. (HTTP implements json handler as default: https://github.com/httprb/http/blob/master/lib/http/mime_type/json.rb)

This problem mostly occurrs when API.ai is down (so should not be a big deal as I hope it is not that often down :D), but still it would be helpful to have a more intuitive exception raised.
Further down in apiai-ruby-cllient the problem is that `fail_or_return_response_body(response.code, response_body)` already takes the parsed response as argument.

apiai-ruby-client should probably only call `#parse` if it knows it's a json response.

Is this anything we should improve? If yes I will try to create a pr for it.

**Comment by saieswar** [comment]
I got this issue when I am  using wrong Client access token 
After giving correct Client access token it was fixed for me

**Comment by saieswar** [comment]
Whe in tested in rails console it is worked fine(text_request)
i tried using ngrok(which gives https url which points to localhost) 
I have 2 machines where if i set webhook url using ngrok on other machine it is working fine. but if i set webhook url on same machine where my rails app exists .it is giving timeout error

**Comment by saieswar** [comment]
I got this error(HTTP::Error Unknown MIME type: text/html) when I used wrong client access token 
and fixed it by giving correct, it may be one of the reasons

**Comment by saieswar** [comment]
correct, but for me sometimes if anything wrong in client access token like adding spaces etc this issue is reproduced we can resolve by correct client access token

**Comment by echan00** [comment]
I just started getting this error from dialogflow, quite concerning as this is breaking our chatbot and all conversations

**Comment by imthatcarlos** [comment]
Just got this error as well.

**Comment by gurschitz** [comment]
Same here, we keep getting this error every once in a while.
